### DISCUSSION TITLE
Does clangd not use -include flags from compile_commands.json?

### DISCUSSION THREAD

**Discussion by thermopyle** [clangd/clangd#1381] [Q&A] [answered]
We use our own version of unity builds where a group of source files are included in a package source file which is visible to the project and compiled. These package source files are then visible in the generated compile_commands.json. Some of the individual source files no longer compiles standalone due to missing includes (includes that are present in the package source files, which makes the project compile as a whole).

I tried to solve this by adding `-include` flags specific for the individual source files that need them, in the `.clangd `configuration file. I can see the inferred commands in the lsp log in Emacs, where these `-include` flags are added to the inferred command. If I copy this command and run it in a shell, compilation works fine. But clangd does not seem to use the `-include` flags itself when it is indexing the individual source files. I get errors and warnings that should have been fixed by the `-include` flags. Does clangd not use `-include `flags present in compile_command.json?

**Comment by HighCommander4** [comment, marked as answer]
I believe they should be used. Can you attach clangd logs?

**Reply by thermopyle** [reply]
I checked the logs and indeed, clangd uses any present `-include` flags. I think I had the paths wrong to the files I tried to include using `-include`, I made them absolute, now I think it works.
